### ISSUE TITLE
Screenshots not showing.

### ISSUE THREAD

**Issue by vijayrawatsan** [openstf/stf#19]
I can take screenshots and see them.
They are being saved in /var/folders/...
But when I change tabs from screenshot to any other tab and come back to screenshot they are gone from screenshots tab(but they are still saved on disk at /var/folders/...).

**Comment by sorccu** [comment]
The screenshot page hasn't received a lot of love so far, although similar persistence issues can be found on all tabs. This is one of the many areas we're planning to improve at some point :(

**Comment by vijayrawatsan** [comment]
@sorccu 
Have you guys implemented this feature and it is not working? Or you have just skipped the feature?
Can you point me in the direction where have you written the code for the same?

**Comment by sorccu** [comment]
The screenshot page has a separate Angular Controller that keeps track of its own state. When you move to a different tab the controller gets destroyed and the state is lost. This is a design issue on our side, all state should persist. The current code is [here](https://github.com/openstf/stf/blob/master/res/app/control-panes/screenshots/screenshots-controller.js), but there's not much to improve there without solving the state issue.

**Comment by vijayrawatsan** [comment]
@sorccu Thanks found it. By state persistence I assume you are talking about client side tab's state. But shouldn't we save the screenshots for a device in db itself so that in the next session user can see his/her screenshots?

**Comment by sorccu** [comment]
Yes, that's one of the many things we're working on, but we're heavily
backlogged. If you feel like helping that would be great too :)

**Comment by vijayrawatsan** [comment]
Lost my js chops a few years back. Don't wanna pollute your code. Anyways, I will discuss with my js friends who might pick this if time allows.
Thanks :)
